core/src/main/java/hudson/PluginManager.java

```diff
@@ -2266,6 +2266,14 @@ public boolean hasAdoptThisPluginLabel(UpdateSite.Plugin plugin) {
         return Arrays.asList(categories).contains("adopt-this-plugin");
     }
 
+    @Restricted(DoNotUse.class) // Used from table.jelly
+    public boolean hasLatestVersionNewerThanOffered(UpdateSite.Plugin plugin) {
+        if (plugin.latest == null) {
+            return false;
+        }
+        return !plugin.latest.equalsIgnoreCase(plugin.version); // we can assume that any defined 'latest' will be newer than the actual offered version
```

Comment:
Isn't this taken care of in the plugin.latest calculation?

Reply:
This code supports `latest` being added to JSON even if the latest release is being offered. While that's not the behavior of update-center2 in the linked PR, I wouldn't consider it wrong.

Reply:
Maybe it's not wrong, but isn't it more sensible to make both sides have the same behaviour? I believe it's easier to undertand.

Reply:
https://en.wikipedia.org/wiki/Robustness_principle